src/Diagnostics.RuntimeHost/Services/ResourceService/StampService.cs

```diff
@@ -46,15 +46,16 @@ public StampService()
 
             var windowsTenantIdsTask = GetTenantIdsAsync(stamp, startTime, endTime, dataProviderContext, PlatformType.Windows);
             var linuxTenantIdsTask = GetTenantIdsAsync(stamp, startTime, endTime, dataProviderContext, PlatformType.Linux);
+            await Task.WhenAll(windowsTenantIdsTask, linuxTenantIdsTask);
             var windowsTenantIds = await windowsTenantIdsTask;
             var linuxTenantIds = await linuxTenantIdsTask;
 
-            PlatformType type = PlatformType.Windows;
             List<string> tenantIds = windowsTenantIds.Union(linuxTenantIds).ToList();
 
+            PlatformType type = PlatformType.Windows | PlatformType.HyperV;
             if (windowsTenantIds.Any() && linuxTenantIds.Any())
             {
-                type = PlatformType.Windows | PlatformType.Linux;
+                type = PlatformType.Windows | PlatformType.HyperV | PlatformType.Linux;
```

Comment:
Can platform be both HyperV and Linux?

Reply:
Ahh nevermind, It can be a megastamp of Linux and Windows machines is it?

Reply:
From the old diff, platform could be both Windows and Linux, so I don't think this code was meant to differentiate between the two.

Reply:
Right, I think we need more information about Stamp behavior with HyperV

Reply:
Guys - We should only set HyperV platform if IsXenon is set to true. Am I missing anything ?

Reply:
I think this infers platform type of the stamp based on the presence of Windows or Linux tenant IDs. So this probably should not be set yet, and I'll get clarification on how to tell if the stamp runs Xenon.